Model/Capabilities.php

```diff
@@ -0,0 +1,34 @@
+<?php
+namespace Omise\Payment\Model;
+
+use Omise\Payment\Model\Api\Capabilities as OmiseCapabilitiesAPI;
+use Omise\Payment\Model\Omise;
+
+class Capabilities
+{
+    /**
+     * @var \Omise\Payment\Model\Omise
+     */
+    protected $omise;
+
+    /**
+     * @var \Omise\Payment\Model\Api\Capabilities
+     */
+    protected $capabilitiesAPI;
+    
+    public function __construct(
+        Omise $omise,
+        OmiseCapabilitiesAPI $capabilitiesAPI
+    ) {
+        $this->omise = $omise;
+        $this->capabilitiesAPI = $capabilitiesAPI;
+
+        $this->omise->defineUserAgent();
+        $this->omise->defineApiVersion();
+        $this->omise->defineApiKeys();
+    }
+
+    public function get() {
```

Comment:
same as `Model\Api\Capabilities` comment.
The method itself doesn't describe its functionality and is ambiguous.

You could have 
```php
// Model\Api\Capabilities
class Capabilities
{
    public function retrieve()
    {
        return OmiseCapabilities::retrieve();
    }
}
```
and 
```php
// Model\Capabilities

class Capabilities
{
    public function retrieveInstallmentBackends()
    {
        $capabilities = $this->capabilitiesAPI->retrieve();
        return $capabilities->getBackends(
	    $capabilities->backendTypeIs('installment')
        );    
    }
}
```

That would be more clear when you execute it at the ConfigProvider class.
```php
public function getConfig()
{
    $storeId = $this->_storeManager->getStore()->getId();
    foreach ($this->_paymentLists->getActiveList($storeId) as $method) {
        if ($method->getCode() === OmiseInstallmentConfig::CODE) {
            return [
                'capabilities' => $this->capabilities->retrieveInstallmentBackends(),
            ];
        }
    }
}
```

Reply:
not sure how future will look like, for now I changed (as we get only from `installment` backend) to getFromInstallments

So
`Capabilities::getFromInstallments` it is.

Reply:
Uhmm.. maybe just my English.. but `getFromInstallments` give a bit different sense to me.
Like, you have `Capabilities` object. And you get `InstallmentBackends`, `InternetBankingBackends`, `CardBackends` **from** the Capabilities.

Thinking in User object case, that contains a various of user informations.
Assumes that you happen to want to get a specific user's firstname to display on the page.
You surely won't call `$user->getFromFirstname` but `$user->getFirstname`.

Same thing to Capabilities object.

Reply:
Or maybe in a more related-case. 
Assume that you wanted to display a purchased-list of a specific user at the user-order-history page. You want to know how many times that this specific user purchased stuff from the store.
You probably go with
```php
foreach($user->purchasedList() as $item) {
    echo "$item->amount - $item->purchased_date";
}
```
or
```php
foreach($user->getPurchasedList() as $item) {
    echo "$item->amount - $item->purchased_date";
}
```

But `getFromPurchasedList()` wouldn't make sense for sure
```php
foreach($user->getFromPurchasedList() as $item) {
    echo "$item->amount - $item->purchased_date";
}
```

Reply:
I'm totally with @guzzilar on this. Reusing the name `capabilities` in a different context from the way it is reference both in `omise-php` and the Capabilities API itself simply introduces unnecessary confusion. Better to stick with the terminology originating from the API.

We are interested in retrieving the properties of a payment `backend` from within the `capabilities` of the merchant account. In this case, this backend is of the `instalment` type. @guzzilar 's suggestions for naming seem very appropriate and communicate well what they do in the context of the Capabilities API

As for `Capabilities::getFromInstallments` or `getFromInstallmentBackend`, to me the names feel linguistically awkward and they depart from conventions in other code that were drawn from those adopted in the API response itself

Reply:
I noticed the same confusing use of `capabilities` in the `getConfig` method - instalment backends are being called 'capabilities'